distribution/test/itests/test-itests-common/src/test/java/ddf/common/test/mock/csw/ChunkedContent.java

```diff
@@ -0,0 +1,158 @@
+import org.glassfish.grizzly.http.server.Response;
+import org.slf4j.LoggerFactory;
+import org.slf4j.ext.XLogger;
+
+import com.xebialabs.restito.semantics.Action;
+import com.xebialabs.restito.semantics.Function;
+
+/**
+ * Encompasses the response for a product retrieval for a Restito stub server.
+ * Allows for the simple setup of specific responses, slow sending, and simulated
+ * network failure retry testing.
+ */
+public class ChunkedContent implements Function<Response, Response> {
+    private static final XLogger LOGGER =
+            new XLogger(LoggerFactory.getLogger(ChunkedContent.class));
+
+    private String responseMessage;
+
+    private long messageDelayMs;
+
+    private int numberOfFailures;
+
+    private int numberOfRetries = 0;
+
+    /**
+     * Constructor for a response that has a delay and a number of planned failure.
+     *
+     * @param responseMessage  Message to be sent in response.
+     * @param messageDelay     Time to wait between sending each character of the message.
+     * @param numberOfFailures Number of times to fail (simulate network disconnect) after sending
+     *                         the first character. Once this number is reached, the message will
+     *                         send successfully.
+     */
+    private ChunkedContent(String responseMessage, Duration messageDelay, int numberOfFailures) {
+        this.responseMessage = responseMessage;
+        this.messageDelayMs = messageDelay.toMillis();
+        this.numberOfFailures = numberOfFailures;
+    }
+
+    /**
+     * Reset states. This will reset any variables used for tracking failure attempts.
+     * This should be called after any complete failure tests to ensure that the response will not
+     * continue waiting for another retry attempt from the calling server.
+     */
+    public void resetFailState() {
+        numberOfRetries = 0;
+    }
+
+    /**
+     * Returns the minimum headers required for sending a chunked data response, but does not
+     * include the response function itself.
+     *
+     * @return the minimum headers required for sending a chunked data response
+     */
+    public static Action getHeaders() {
+        return composite(contentType("text/plain"),
+                header("Transfer-Encoding", "chunked"),
+                header("content-type", "text/plain"));
+    }
+
+    /**
+     * Constructor for a response that has a delay and a number of planned failure.
+     *
+     * @param responseMessage  Message to be sent in response.
+     * @param messageDelay     Time to wait between sending each character of the message.
+     * @param numberOfFailures Number of times to fail (simulate network disconnect) after sending
+     *                         the first character. Once this number is reached, the message will
+     *                         send successfully.
+     */
+    public static Action newResponse(String responseMessage, Duration messageDelay,
```

Comment:
As I mentioned in my previous comments, I would try to stay as close to the `Action` naming convention as possible and call these methods `chunkedContent`. I know it looks a little redundant, but it looks better if you use static imports. Here's an example:

```
import static ddf.common.test.restito.ChunkedResponse.chunkedResponse;

@Test
public void test() {
    whenHttp(server).match(...).then(chunkedContent(...));
}
```